### PR TITLE
expression: try to fix TestExprPushDownToFlash tests

### PR DESCRIPTION
**Pull request by xhebox**
### What problem does this PR solve?

Problem Summary: So `TestExprPushDownToFlash` failed on my local machine. It is unstable and hard to reproduce. The only questionable code is the failpoint in the `CanPushDownToXXX`, which is enabled by the global `func init()`. I don't know if it is caused by goroutine schedule, forking, race, execution order or whatever.

It is moved into the entry point `TestT`, together with another `func init()`. And all tests passed on my machine, so I think this modification is fine.

FYI, the introduction PR for failpoint and timeutil is #15174, #14826.

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- No release note

### CODE REVIEW

File: expression/evaluator_test.go [hunk 1]
```diff
@@ -48,7 +50,21 @@ func TestT(t *testing.T) {
...
+	fpname := "github.com/pingcap/tidb/expression/PanicIfPbCodeUnspecified"
+	err := failpoint.Enable(fpname, "return(true)")
+	if err != nil {
+		t.Fatalf("enable global failpoint `%s` failed: %v", fpname, err)
+	}
+	// Some test depends on the values of timeutil.SystemLocation()
+	// If we don't SetSystemTZ() here, the value would change unpredictable.
+	// Affectd by the order whether a testsuite runs before or after integration test.
+	// Note, SetSystemTZ() is a sync.Once operation.
+	timeutil.SetSystemTZ("system")
```
[the remark this answers is not in the document]
Comment (xhebox) [reply expression/evaluator_test.go:62]: Done.

**Review by AilinKid — COMMENTED** [review]
lgtm

**Pull request merged**